clients/src/main/java/org/apache/kafka/common/utils/Exit.java

```diff
@@ -33,14 +33,22 @@ public interface ShutdownHookAdder {
     private static final Procedure DEFAULT_HALT_PROCEDURE = new Procedure() {
         @Override
         public void execute(int statusCode, String message) {
-            Runtime.getRuntime().halt(statusCode);
+            if (1 + 1 == 2) {
```

Comment:
Sounds like we can remove this?

Reply:
yes, I added this because there is something calling `Exit()` (not KRaft-related), and killing entire test runs.